torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,389 @@
+
+    def __len__(self) -> int:
+        """Return the number of items in the dataset.
+
+        Returns:
+            length of dataset in integer
+        """
+        return len(self.chip_paths)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Returns a sample from dataset.
+
+        Args:
+            Index: index to return
+
+        Returns:
+            data and label at given index
+        """
+        image = self._load_image(index)
+        label = self._load_target(index)
+        sample: Dict[str, Tensor] = {"image": image, "mask": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load all source images for a chip.
+
+        Args:
+            index: position of the indexed chip
+
+        Returns:
+            a tensor of stacked source image data
+        """
+        source_asset_paths = self.chip_paths[index]["source"]
+        images = []
+        for path in source_asset_paths:
+            with rasterio.open(path) as image_data:
+                image_array = image_data.read(1).astype(np.int32)
+                images.append(image_array)
+        image_stack: "np.typing.NDArray[np.int_]" = np.stack(images, axis=0)
+        image_tensor = torch.from_numpy(image_stack)
+        return image_tensor
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load label image for a chip.
+
+        Args:
+            index: position of the indexed chip
+
+        Returns:
+            a tensor of the label image data
+        """
+        label_asset_path = self.chip_paths[index]["target"][0]
+        with rasterio.open(label_asset_path) as target_data:
+            target_img = target_data.read(1).astype(np.int32)
+
+        target_array: "np.typing.NDArray[np.int_]" = np.array(target_img)
+        target_tensor = torch.from_numpy(target_array)
+        return target_tensor
+
+    @staticmethod
+    def _read_json_data(object_path: str) -> Any:
+        with open(object_path) as read_contents:
+            json_data = json.load(read_contents)
+        return json_data
+
+    def _load_items(self, item_json: str) -> Dict[str, List[str]]:
+        """Loads the label item and corresponding source items.
+
+        Args:
+            item_json: a string path to the item JSON file on disk
+
+        Returns:
+            a dictionary with paths to the source and target TIF filenames
+        """
+        item_meta = {}
+
+        label_data = self._read_json_data(item_json)
+        label_asset_path = os.path.join(
+            os.path.split(item_json)[0], label_data["assets"]["labels"]["href"]
+        )
+        item_meta["target"] = [label_asset_path]
+
+        source_item_hrefs = [
+            os.path.join(self.root, link["href"].replace("../../", ""))
+            for link in label_data["links"]
+            if link["rel"] == "source"
+        ]
```

Comment:
A lot of these list comprehensions would probably be more legible as for loops. I rarely use list comprehensions unless I can fit everything on one line.

Reply:
Swapped all the list compressions in `cloud_cover.py` to for loops, hope I didn't' break anything :-D